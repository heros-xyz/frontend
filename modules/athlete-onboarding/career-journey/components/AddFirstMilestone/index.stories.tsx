import type { Meta, StoryObj } from "@storybook/react";
import AddFirstMilestone from ".";

const meta: Meta<typeof AddFirstMilestone> = {
  title: "Athlete/AddFirstMilestone",
  component: AddFirstMilestone,
  tags: ["autodocs"],
  parameters: {
    layout: "xl",
  },
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof AddFirstMilestone>;

export const AddFirstMilestone1: Story = {};
