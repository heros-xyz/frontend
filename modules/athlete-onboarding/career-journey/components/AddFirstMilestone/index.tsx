import { Box, Button, Text } from "@chakra-ui/react";
import { FirstloadIcon } from "@/components/svg/FirstloadIcon";
import AddArchiveIcon from "@/components/svg/AddArchiveIcon";
import HerosOnboardingWrapperNew from "@/components/ui/HerosOnboardingWrapperNew";
import { IconArrowRight } from "@/components/svg/IconArrowRight";

interface AddFirstMilestoneProps {
  handleAddMilestone: () => void;
}

const AddFirstMilestone: React.FC<AddFirstMilestoneProps> = ({
  handleAddMilestone,
}) => {
  return (
    <HerosOnboardingWrapperNew
      Icon={
        <FirstloadIcon
          w={{ base: "150px", xl: "240px" }}
          h={{ base: "150px", xl: "240px" }}
          color={{ base: "accent.1", xl: "white" }}
        />
      }
      textButton="Proceed"
      IconButton={<IconArrowRight />}
      title="Career journey"
      bgIconColor="accent.1"
      isDisabled
    >
      <Box>
        <Box mb={2}>
          <Text
            mt={{ base: 10, sm: 0 }}
            fontSize={{ xl: "medium" }}
            color="grey.300"
          >
            Try adding a milestone. It can be the day you joined a team, the day
            you won a prize, or even the time period you trained at a facility.
            Tell your journey!
          </Text>
        </Box>
        <Box
          display="flex"
          verticalAlign="middle"
          alignItems="center"
          mt="3.5rem"
          ml={{ base: 10, xl: 2 }}
          color="primary"
        >
          <Button
            p={0}
            bg="transparent"
            _hover={{ bg: "transparent" }}
            _active={{ bg: "transparent" }}
            onClick={handleAddMilestone}
          >
            <AddArchiveIcon
              w={{ base: "30px", xl: "45px" }}
              h={{ base: "30px", xl: "45px" }}
            />
          </Button>
          <Text
            as="span"
            fontWeight={"medium"}
            ml={2}
            fontSize={{ xl: "2rem" }}
          >
            Add your first milestone
          </Text>
        </Box>
      </Box>
    </HerosOnboardingWrapperNew>
  );
};

export default AddFirstMilestone;
