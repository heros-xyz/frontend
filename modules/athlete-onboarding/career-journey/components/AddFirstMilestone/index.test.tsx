import { render } from "@testing-library/react";
import AddFirstMilestone from ".";

it("renders homepage unchanged", () => {
  const { container } = render(<AddFirstMilestone isActive />);
  expect(container).toMatchSnapshot();
});
