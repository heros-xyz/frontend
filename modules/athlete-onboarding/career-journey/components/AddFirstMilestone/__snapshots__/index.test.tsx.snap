// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-1jvt6v8"
  >
    <div
      class="css-1kmmdcq"
    >
      Career journey
    </div>
    <div
      class="chakra-container css-17f3xev"
    >
      <div
        class="css-1ispe57"
      >
        <div
          class="css-1tvidti"
        >
          <div
            class="css-16g2rey"
          >
            <svg
              class="chakra-icon css-1i9wppb"
              focusable="false"
              viewBox="0 0 151 150"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="75.5"
                cy="75"
                fill="currentColor"
                r="74.5"
                stroke="#DDDDDD"
              />
              <g
                clip-path="url(#clip0_1504_20116)"
              >
                <path
                  clip-rule="evenodd"
                  d="M88.3008 93.6L90.5008 74C93.4008 75.8 98.2008 84.8 99.8008 88.7C101.801 93.9 100.701 101.7 99.6008 107.4C98.5008 106.8 97.2008 106.4 95.8008 106.4C95.3008 106.4 94.7008 106.5 94.2008 106.6C93.9008 101.1 91.3008 98.1 88.3008 93.6Z"
                  fill="#CDEAEC"
                  fill-rule="evenodd"
                />
                <path
                  d="M88.3008 93.6L90.5008 74C93.4008 75.8 98.2008 84.8 99.8008 88.7C101.801 93.9 100.701 101.7 99.6008 107.4C98.5008 106.8 97.2008 106.4 95.8008 106.4C95.3008 106.4 94.7008 106.5 94.2008 106.6C93.9008 101.1 91.3008 98.1 88.3008 93.6Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M88.3008 93.6L90.5008 74C93.4008 75.8 98.2008 84.8 99.8008 88.7C101.801 93.9 100.701 101.7 99.6008 107.4C98.5008 106.8 97.2008 106.4 95.8008 106.4C95.3008 106.4 94.7008 106.5 94.2008 106.6C93.9008 101.1 91.3008 98.1 88.3008 93.6Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clip-rule="evenodd"
                  d="M62.0001 93.6L59.7001 74C56.9001 75.8 52.0001 84.8 50.5001 88.7C48.3001 94.3 49.8001 102.9 51.0001 108.6C51.6001 108.4 52.2001 108.4 52.9001 108.4C54.0001 108.4 55.1001 108.6 56.1001 109.1C55.6001 101.9 58.5001 98.7 62.0001 93.6Z"
                  fill="#CDEAEC"
                  fill-rule="evenodd"
                />
                <path
                  d="M62.0001 93.6L59.7001 74C56.9001 75.8 52.0001 84.8 50.5001 88.7C48.3001 94.3 49.8001 102.9 51.0001 108.6C51.6001 108.4 52.2001 108.4 52.9001 108.4C54.0001 108.4 55.1001 108.6 56.1001 109.1C55.6001 101.9 58.5001 98.7 62.0001 93.6Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M62.0001 93.6L59.7001 74C56.9001 75.8 52.0001 84.8 50.5001 88.7C48.3001 94.3 49.8001 102.9 51.0001 108.6C51.6001 108.4 52.2001 108.4 52.9001 108.4C54.0001 108.4 55.1001 108.6 56.1001 109.1C55.6001 101.9 58.5001 98.7 62.0001 93.6Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M67.9012 82.7L67.7012 112.3"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M74.2012 100V109.7"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M80 96.3V102.3"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M46.3003 118.4C45.4003 116.9 45.2003 112.3 48.1003 109.9C50.0003 108.3 53.7003 107.7 57.7003 109.8L58.1003 110.1"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M58.3008 110C59.1008 106.5 61.5008 104.7 64.9008 105.2"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M83.5 103.1C88.7 104 90 107.4 89.5 109.8"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M92.8008 107.6C100.001 103.8 105.801 111.9 102.601 116.9"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clip-rule="evenodd"
                  d="M87.8995 95.2C89.0995 92 90.6995 79.1 90.5995 70.6C90.2995 46.4 77.3995 30.8 74.8995 31.8C71.4995 33.1 59.7995 50.6 59.3995 70.3C59.1995 80.5 61.3995 90.5 61.9995 95.2C61.9995 95.2 86.7995 95.3 87.8995 95.2Z"
                  fill="#F0C2DA"
                  fill-rule="evenodd"
                />
                <path
                  d="M87.8995 95.2C89.0995 92 90.6995 79.1 90.5995 70.6C90.2995 46.4 77.3995 30.8 74.8995 31.8C71.4995 33.1 59.7995 50.6 59.3995 70.3C59.1995 80.5 61.3995 90.5 61.9995 95.2C61.9995 95.2 86.7995 95.3 87.8995 95.2Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M87.8995 95.2C89.0995 92 90.6995 79.1 90.5995 70.6C90.2995 46.4 77.3995 30.8 74.8995 31.8C71.4995 33.1 59.7995 50.6 59.3995 70.3C59.1995 80.5 61.3995 90.5 61.9995 95.2C61.9995 95.2 86.7995 95.3 87.8995 95.2Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clip-rule="evenodd"
                  d="M66.2012 44.2001C70.0012 36.0001 74.6012 31.5001 75.9012 32.0001C77.6012 32.6001 81.4012 37.3001 84.7012 44.2001H66.2012Z"
                  fill="#ED3338"
                  fill-rule="evenodd"
                />
                <path
                  d="M66.2012 44.2001C70.0012 36.0001 74.6012 31.5001 75.9012 32.0001C77.6012 32.6001 81.4012 37.3001 84.7012 44.2001H66.2012Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M66.2012 44.2001C70.0012 36.0001 74.6012 31.5001 75.9012 32.0001C77.6012 32.6001 81.4012 37.3001 84.7012 44.2001H66.2012Z"
                  fill="#5C54A4"
                />
                <path
                  d="M66.2012 44.2001C70.0012 36.0001 74.6012 31.5001 75.9012 32.0001C77.6012 32.6001 81.4012 37.3001 84.7012 44.2001H66.2012Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clip-rule="evenodd"
                  d="M75.4004 56.8C79.0004 56.8 81.9004 59.7 81.9004 63.3C81.9004 66.9 79.0004 69.8 75.4004 69.8C71.8004 69.8 68.9004 66.9 68.9004 63.3C68.9004 59.7 71.8004 56.8 75.4004 56.8Z"
                  fill="#D7E037"
                  fill-rule="evenodd"
                />
                <path
                  d="M75.4004 56.8C79.0004 56.8 81.9004 59.7 81.9004 63.3C81.9004 66.9 79.0004 69.8 75.4004 69.8C71.8004 69.8 68.9004 66.9 68.9004 63.3C68.9004 59.7 71.8004 56.8 75.4004 56.8Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M75.4004 56.8C79.0004 56.8 81.9004 59.7 81.9004 63.3C81.9004 66.9 79.0004 69.8 75.4004 69.8C71.8004 69.8 68.9004 66.9 68.9004 63.3C68.9004 59.7 71.8004 56.8 75.4004 56.8Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_1504_20116"
                >
                  <rect
                    fill="white"
                    height="87.8"
                    transform="translate(45 31)"
                    width="59.2"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
        <div
          class="css-14s5ckc"
        >
          <div
            class="css-1s2o82u"
          >
            <div
              class="css-1uoge8t"
            >
              <div
                class="css-0"
              >
                <div
                  class="css-wlea3r"
                >
                  <p
                    class="chakra-text css-ocgwbc"
                  >
                    Try adding a milestone. It can be the day you joined a team, the day you won a prize, or even the time period you trained at a facility. Tell your journey!
                  </p>
                </div>
                <div
                  class="css-dbol0d"
                >
                  <button
                    class="chakra-button css-1g2mshn"
                    type="button"
                  >
                    <svg
                      class="chakra-icon css-17cusq9"
                      focusable="false"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M31 16C31 24.2843 24.2843 31 16 31C7.71573 31 1 24.2843 1 16C1 7.71573 7.71573 1 16 1C24.2843 1 31 7.71573 31 16Z"
                      />
                      <path
                        d="M16 11V16M16 16V21M16 16H21M16 16H11M31 16C31 24.2843 24.2843 31 16 31C7.71573 31 1 24.2843 1 16C1 7.71573 7.71573 1 16 1C24.2843 1 31 7.71573 31 16Z"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </button>
                  <span
                    class="chakra-text css-67xhbk"
                  >
                    Add your first milestone
                  </span>
                </div>
              </div>
              <div
                class="css-lrby83"
              >
                <button
                  class="chakra-button css-k6pd2s"
                  type="button"
                >
                  Proceed
                  <span
                    class="css-9n3bbc"
                  >
                    <svg
                      class="chakra-icon css-nyt2fo"
                      focusable="false"
                      viewBox="0 0 12 11"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1 5.5L11 5.5M11 5.5L6.5 1M11 5.5L6.5 10"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-1mv50kv"
            disabled=""
            type="button"
          >
            Proceed
            <span
              class="css-9n3bbc"
            >
              <svg
                class="chakra-icon css-nyt2fo"
                focusable="false"
                viewBox="0 0 12 11"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 5.5L11 5.5M11 5.5L6.5 1M11 5.5L6.5 10"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
